lib/rubocop/cop/lint/struct_new_override.rb

```diff
@@ -0,0 +1,50 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Lint
+      # This cop checks unexpected overrides of the `Struct` original methods
+      # via `Struct.new`.
+      #
+      # @example
+      #   # bad
+      #   Bad = Struct.new(:members, :clone, :count)
+      #   b = Bad.new([], true, 1)
+      #   b.members #=> [] (overriding `Struct#members`)
+      #   b.clone #=> true (overriding `Object#clone`)
+      #   b.count #=> 1 (overriding `Enumerable#count`)
+      #
+      #   # good
+      #   Good = Struct.new(:id, :name)
+      #   g = Good.new(1, "foo")
+      #   g.members #=> [:id, :name]
+      #   g.clone #=> #<struct Good id=1, name="foo">
+      #   g.count #=> 2
+      #
+      class StructNewOverride < Cop
+        MSG = 'Disallow overriding the `Struct#%<method_name>s` method.'
+
+        STRUCT_METHOD_NAMES = Struct.instance_methods
+
+        def_node_matcher :struct_new, <<~PATTERN
+          (send
+            (const ${nil? cbase} :Struct) :new ...)
+        PATTERN
+
+        def on_send(node)
+          return unless struct_new(node) do
+            node.arguments.each do |arg|
+              next unless arg.respond_to?(:value)
+
+              method_name = arg.value
```

Comment:
I guess it needs `to_sym` because the method name may be a String.

```ruby
# Both of :foo and 'bar' are method names.
Struct.new(:foo, 'bar')
```

Reply:
Oh, I couldn't notice this spec of `Struct.new` to allow string member names! 😳 
I've addressed via cb46c1a1c2204e7b0ce94ae8c6074a2b85133e94. Thank you!

---

By the way, we may be able to add a new cop to check the usage of string member names. 😅 (Or, we need a type checker...?)
Because it seems that string member names can be allowed only for backward compatibility.